lxd/main_cluster.go

```diff
@@ -0,0 +1,158 @@
+	if err != nil {
+		return errors.Wrapf(err, "Failed to open local database.")
+	}
+
+	addresses, err := cluster.ListDatabaseNodes(db)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get database nodes.")
+	}
+
+	printDatabaseNodes(addresses)
+
+	return nil
+}
+
+func printDatabaseNodes(addresses []string) {
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetAlignment(tablewriter.ALIGN_LEFT)
+	table.SetAutoWrapText(false)
+	table.SetAutoFormatHeaders(false)
+	table.SetHeader([]string{"Address"})
+	for _, address := range addresses {
+		data := []string{address}
+		table.Append(data)
+	}
+	table.Render()
+}
+
+type cmdClusterRecoverFromQuorumLoss struct {
+	global             *cmdGlobal
+	flagNonInteractive bool
+}
+
+func (c *cmdClusterRecoverFromQuorumLoss) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = i18n.G("recover-from-quorum-loss")
+	cmd.Aliases = []string{"ls"}
+	cmd.Short = i18n.G("Recover a LXD instance whose cluster has lost quorum")
+
+	cmd.RunE = c.Run
+
+	cmd.Flags().BoolVarP(&c.flagNonInteractive, "quiet", "q", false, i18n.G("Don't require user confirmation"))
+
+	return cmd
+}
+
+func (c *cmdClusterRecoverFromQuorumLoss) Run(cmd *cobra.Command, args []string) error {
+	// Make sure that the daemon is not running.
+	_, err := lxd.ConnectLXDUnix("", nil)
+	if err == nil {
+		return fmt.Errorf("The LXD daemon is running, please stop it first.")
+	}
+
+	// Prompt for confiromation unless --quite was passed..
+	if !c.flagNonInteractive {
+		err := c.promptConfirmation()
+		if err != nil {
+			return err
+		}
+	}
+
+	os := sys.DefaultOS()
+
+	db, _, err := db.OpenNode(filepath.Join(os.VarDir, "database"), nil, nil)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to open local database.")
+	}
+
+	return cluster.Recover(db)
+}
+
+func (c *cmdClusterRecoverFromQuorumLoss) promptConfirmation() error {
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Printf(i18n.G(`You should run this command only if you are *absolutely* certain that this is
+the only database node left in your cluster AND that other database nodes will
+never come back (i.e. their LXD daemon won't ever be started again).
+
+This will make this LXD instance the only member of the cluster, and it won't
+be possible to perform operations on former cluster members anymore.
+
+However all information about former cluster members will be preserved in the
+database, so you can possibly inspect it for further recovery.
+
+You'll be able to permanently delete from the database all information about
+former cluster members by running "lxc cluster remove <member-name> --force".
+
+See https://lxd.readthedocs.io/en/latest/clustering/#disaster-recovery for more
+info.
+
+Do you want to proceed? (yes/no): `))
+	input, _ := reader.ReadString('\n')
+	input = strings.TrimSuffix(input, "\n")
+
+	if !shared.StringInSlice(strings.ToLower(input), []string{i18n.G("yes")}) {
+		return fmt.Errorf(i18n.G("User aborted delete operation"))
```

Comment:
Error needs changing, also, we don't use `i18n` inside the `lxd` binary.

Reply:
Fixed.